### DISCUSSION TITLE
More flexibility for `seekTo`

### DISCUSSION THREAD

**Discussion by ffxsam** [katspaugh/wavesurfer.js#3403] [Ideas]
It would be cool if `seekTo` could be used to jump to either a percentage (progress) of a track, or an exact time (in seconds). I think we could introduce this in a non-breaking manner:

```ts
type SeekToPerc = { progress: number; secs?: never; }
type SeekToSecs = { progress?: never; secs: number; }
type SeekToParams = number | (SeekToPerc | SeekToSecs);

/**
 * seekTo function where user can pass either a number, or an options object where EITHER progress or secs can be specified, but not both.
 */
function seekTo(percOrOptions: SeekToParams) {
    //
}

seekTo(23);
seekTo({
    progress: 0.37,
    secs: 213,
}); // invalid
seekTo({
    secs: 3,
    progress: 0
}); // invalid
seekTo({}); // invalid
seekTo({ progress: 0.7 }); // OK
seekTo({ secs: 213 }); // OK
```
The benefit here is that if a user already has seconds and nothing else, they don't have to get the total duration and divide into it just to seek.

Happy to write this and open a PR!

**Comment by katspaugh** [comment]
There's technically also `setTime` which takes seconds.

**Comment by ffxsam** [comment]
D'oh - thanks, I missed that among the rest of the methods! Closing.

<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExeGw4azlzZjNuMmo5cWZ3YjhndXNxazRnemN2dngwaTg3dTUwdDIyNyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/a93jwI0wkWTQs/giphy.gif">
